### PR TITLE
Add tests for Intl Enumeration API

### LINKED ISSUE #3131: Need test for Intl Enumeration API
Intl Enumeration API reach Stage 3 in July 2021 TC39 meeting. file issue to track the tests

https://tc39.es/proposal-intl-enumeration/
https://github.com/tc39/proposal-intl-enumeration

### PR DESCRIPTION
**Pull request by anba**
Covers the usual surface tests and additional functionality tests which were
upstreamed from existing tests in SpiderMonkey.

Fixes #3131

### CODE REVIEW

**Comment by rwaldron** [comment]
@anba three lint issues: 

```
/home/circleci/test262/test/intl402/Intl/supportedValuesOf/builtin.js: FRONTMATTER - Unrecognized fields: include
/home/circleci/test262/test/intl402/Intl/supportedValuesOf/numberingSystems-with-simple-digit-mappings.js: INCLUDES - Unused include: "testIntl.js"
/home/circleci/test262/test/intl402/NumberFormat/prototype/format/numbering-systems.js: INCLUDES - Unused include: "testIntl.js"
```

**Review by rwaldron — CHANGES_REQUESTED** [review]
@anba three lint issues: 

```
/home/circleci/test262/test/intl402/Intl/supportedValuesOf/builtin.js: FRONTMATTER - Unrecognized fields: include
/home/circleci/test262/test/intl402/Intl/supportedValuesOf/numberingSystems-with-simple-digit-mappings.js: INCLUDES - Unused include: "testIntl.js"
/home/circleci/test262/test/intl402/NumberFormat/prototype/format/numbering-systems.js: INCLUDES - Unused include: "testIntl.js"
```

File: test/intl402/Intl/supportedValuesOf/numberingSystems-with-simple-digit-mappings.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  AvailableNumberingSystems ( )
+    The AvailableNumberingSystems abstract operation returns a List, ordered as
+    if an Array of the same values had been sorted using %Array.prototype.sort%
+    using undefined as comparefn, that contains unique numbering systems
+    identifiers identifying the numbering systems for which the implementation
+    provides the functionality of Intl.DateTimeFormat, Intl.NumberFormat, and
+    Intl.RelativeTimeFormat objects. The list must include the Numbering System
+    value of every row of Table 4, except the header row.
+includes: [testIntl.js]
```
Comment (leobalter) [inline test/intl402/Intl/supportedValuesOf/numberingSystems-with-simple-digit-mappings.js:26]:
```suggestion
```

File: test/intl402/NumberFormat/prototype/format/numbering-systems.js [hunk 2]
```diff
@@ -7,93 +7,17 @@ description: >
     Tests that Intl.NumberFormat.prototype.format supports all
     numbering systems with simple digit mappings.
 author: Roozbeh Pournader
+includes: [testIntl.js]
```
Comment (leobalter) [inline test/intl402/NumberFormat/prototype/format/numbering-systems.js:10]:
```suggestion
```

File: test/intl402/Intl/supportedValuesOf/builtin.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    Unless otherwise specified every built-in function and every built-in
+    constructor has the Function prototype object, which is the initial value
+    of the expression Function.prototype (20.2.3), as the value of its
+    [[Prototype]] internal slot.
+
+    Built-in function objects that are not identified as constructors do not
+    implement the [[Construct]] internal method unless otherwise specified in
+    the description of a particular function.
+include: [isConstructor.js]
```
Comment (leobalter) [inline test/intl402/Intl/supportedValuesOf/builtin.js:25]:
```suggestion
includes: [isConstructor.js]
```

**Review by leobalter — COMMENTED** [review]
Adding the suggestions here that might help speeding up the process to land this PR.

**Comment by anba** [comment]
Updates:
- 9e88c32d067a65e40060ae0625386e76e1023197
Add `numberingSystemDigits` to the list of exported objects. This resolves the unused include warnings for "testIntl.js".

- 069f2e8a9eed63604795426e89dae4566bdfd0b7
Fixed `include` &rarr; `includes` typo. And then added "Reflect.construct" to the features list to avoid a new linter warning.

**Review by rwaldron — APPROVED** [review]

**Pull request merged**